internal/pkg/cache/redis/redis.go

```diff
@@ -40,16 +45,30 @@ func NewRedis(address string) *Redis {
 
 // NewRedisWithConfig creates a new Redis instance with configurable options.
 func NewRedisWithConfig(cfg *config.RedisConfig) *Redis {
-	o := &redis.UniversalOptions{
-		Addrs:           parseRedisAddress(cfg.Address),
-		MinIdleConns:    cfg.MinIdleConns,
-		PoolSize:        cfg.PoolSize,
-		ConnMaxIdleTime: cfg.IdleTimeout,
-		ConnMaxLifetime: cfg.MaxConnAge,
+	var c redis.UniversalClient
+
+	if isRedisClusterAddress(cfg.Address) {
```

Comment:
Please use a different config variable to indicate to the service whether or not the redis server is in cluster mode or not.